### PR TITLE
fix: eliminate exog-lag scan carry to fix panel lag(x) logp bug

### LINKED ISSUE #316: getting 0 posterior for lag effect
I simulate data as 

``` 
ngeo = 5
ntime = 10
data = pd.DataFrame()
for g in range(ngeo):
  spend = np.random.normal(10, 1, ntime)
  sales = np.ones(ntime) * 20
  sales[1:] = 10 + spend[:-1] + np.random.normal(0, 1, ntime-1)
  df = pd.DataFrame({"spend": spend, "sales": sales})
  df['week'] = np.arange(ntime)
  df['geo'] = g
  data = pd.concat((data, df))
```

I then estimate

``` 
model = pathmc.model(
        "sales ~ lag(spend)",
        data=data,
        panel={"unit": "geo", "time": "week"},
        pooling="partial",
    )
model.fit(draws=1000, chains=2)
``` 

But I get a posterior 0 for lag(spend), why? True value is 1.

### PR DESCRIPTION
**Pull request by drbenvincent**
## Summary

Fixes the root-cause bug behind #316: panel models using `lag(x)` (exogenous lag) returned zero posteriors for the lag coefficient because `pm.sample`'s gradient computation evaluated against the wrong mu.

## Root Cause

PyTensor's scan-merge optimizer incorrectly permuted sit_sot carry channels when `pytensor.function` compiled `[mu_det, logp]` together. This happened because the exogenous lag carry had a **trivially-echo update** (`carry[t+1] = sequence[t]`), making two structurally-identical scan nodes (one from `replace_rvs_by_values`, one from `model.logp()`) candidates for merging. The merge incorrectly swapped the spend-carry with the sales-carry, producing:

```
mu_sales[t≥1] = intercept + beta * mu_sales[t-1]   # WRONG
```
instead of:
```
mu_sales[t≥1] = intercept + beta * spend[t-1]       # CORRECT
```

## Fix

Pre-compute lagged exogenous sequences as tensors **before** the scan:
```python
lagged_seq = pt.concatenate([init_row[None, :], exog_data[:-1]], axis=0)
```

These tensors are derived from `pm.Data` nodes, so `pm.set_data()` and `do()` interventions continue to propagate. With no trivial-echo carry, the optimizer has nothing to incorrectly merge.

## Testing

The 4 `xfail(strict=True)` red tests installed by #328 now pass unconditionally:

- `test_model_consistent[panel-lag(x)-complete]` ✅
- `test_model_consistent[panel-lag(x)-partial]` ✅
- `test_gaussian_likelihood_matches_hand_oracle[panel-lag(x)-complete]` ✅
- `test_gaussian_likelihood_matches_hand_oracle[panel-lag(x)-partial]` ✅

Full suite: 772 passed, 0 failures. Lint: clean.

Closes #316
Related: #326, #327

### CODE REVIEW

**Comment by drbenvincent** [comment]
Should be ready to merge if checks pass

**Review by juanitorduz — APPROVED** [review]
